### PR TITLE
Mesh

### PR DESCRIPTION
**Pull request by tomkenmag**
MeshPatt class finally been moved over with docstrings, tests and all that stuff.

### CODE REVIEW

File: permuta/MeshPatt.py [hunk 1]
```diff
@@ -272,16 +265,16 @@ def shade(self, positions):
 
         return MeshPatt(self.pattern, self.shading | positions)
 
-    def add_point(self, box, shade_dir=DIR_NONE, safe=True):
-        """Returns a mesh pattern with a point added in box.  If shade_dir is
+    def add_point(self, pos, shade_dir=DIR_NONE, safe=True):
```
Comment (RagnarArdal) [inline permuta/MeshPatt.py]: We need to eventually go through and refactor the variable names like we did with Perm last year. I don't think that time is now though, but it needs to happen eventually.

File: permuta/MeshPatt.py [hunk 2]
```diff
@@ -338,45 +331,83 @@ def add_point(self, box, shade_dir=DIR_NONE, safe=True):
...
+            l is a list of integer indices of the permutation perm such that
+            self.pattern == permuta.Perm.to_standard([perm[i] for i in l])
+            and that no element not in the occurrence falls within self.shading.
+        """
+        # TODO: Implement all nice
+        indices = list(range(len(perm)))
+        for candidate_indices in itertools.combinations(indices, len(self)):
+            candidate = [perm[index] for index in candidate_indices]
+            if Perm.to_standard(candidate) != self.pattern:
+                continue
```
Comment (RagnarArdal) [inline permuta/MeshPatt.py]: This should at least use the occurrence code of Perm. We don't want this piece of code to be this dumb at the master branch, and I say that having written this code myself.

**Comment by RagnarArdal** [comment]
Yes, just change the occurrence code and we can merge :+1:

File: permuta/MeshPatt.py (hunk 1 above)
Comment (tomkenmag) [inline permuta/MeshPatt.py]: There is a bunch of stuff that we should refactor, including variable names. Function names seem to be mostly ok, but we might also want to reconsider the format of the docstrings.

**Pull request merged**